google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/data/v2/BigtableDataClient.java

```diff
@@ -235,7 +240,69 @@ public boolean exists(String tableId, String rowKey) {
    * @throws com.google.api.gax.rpc.ApiException when a serverside error occurs
    */
   public boolean exists(String tableId, ByteString rowKey) {
-    return ApiExceptions.callAndTranslateApiException(existsAsync(tableId, rowKey));
+    return exists(tableId, rowKey, new ExistsOptions());
+  }
+
+  /**
+   * Confirms synchronously if given row key exists or not. This method allows customized checking
+   * options by passing in a {@link ExistsOptions}.
+   *
+   * <p>Sample code:
+   *
+   * <pre>{@code
+   * try (BigtableDataClient bigtableDataClient = BigtableDataClient.create("[PROJECT]", "[INSTANCE]")) {
+   *   String tableId = "[TABLE]";
+   *   String key = "key";
+   *
+   *   String authorizedViewId = "[AUTHORIZED_VIEW]";
+   *   ExistsOptions existsOptions = new ExistsOptions().authorizedView(authorizedViewId);
+   *
+   *   boolean isRowPresent = bigtableDataClient.exists(tableId, key, existsOptions);
+   *
+   *   // Do something with result, for example, display a message
+   *   if(isRowPresent) {
+   *     System.out.println(key + " is present");
+   *   }
+   * } catch(ApiException e) {
+   *   e.printStackTrace();
+   * }
+   * }</pre>
+   *
+   * @throws com.google.api.gax.rpc.ApiException when a serverside error occurs
+   */
+  public boolean exists(String tableId, String rowKey, @Nullable ExistsOptions existsOptions) {
```

Comment:
Here and everywhere, do we need the `@Nullable` annotation here? looks like other methods are calling with an empty instance of the existOption?

Reply:
I think potentially this annotation won't do any harm as well?

Reply:
I think this @Nullable annotation enables users to call this function like `exists("my-table", "row1", nullptr)`. Let me know if you don't think we should allow this type of usage and I'll remove the annotation.

Reply:
This does allows the users to call the API in a uniformed manner though, regardless of whether they have an initialized `ExistsOptions` or it's a `nullptr`. Therefore, users can simply call this API without needing to differentiate the two cases with an `if` statement.

Also I found that we have similar APIs with the `Nullable` annotation like the following:
```
public Row readRow(String tableId, String rowKey, @Nullable Filter filter) {
    return readRow(tableId, rowKey, ReadRowOptions.create().filter(filter));
}
```

What do you think?

Reply:
But I think the caller should differentiate the null vs has a viewId case? Because we do expect different results for the 2 cases?

Reply:
Done.